### PR TITLE
workaround for 1 elem hists

### PR DESCRIPTION
**Pull request by SimonDanisch**
`hist([8,8,8,8,8])` would yield histograms with nothing.
With this PR, they get one bar at 8:
image

Not sure if it's the nicest way to set edges to nothing in that case, but since the edges are kind of undefined in that case, most other values would not work out?

### CODE REVIEW

File: src/stats/hist.jl [hunk 1]
```diff
@@ -72,6 +72,9 @@ function Makie.plot!(plot::Hist)
     edges = lift(values, plot.bins) do vals, bins
         if bins isa Int
             mi, ma = float.(extrema(vals))
+            if mi == ma
+                return nothing
```
Comment (piever) [inline src/stats/hist.jl:76]: Can't we just set

```julia
mi, ma = mi - 0.5, ma + 0m5
```

and keep all the remaining logic the same? We still get the arbitrary bar of width `1` centered around the unique value, but avoid having to keep two different code paths.

In particular, I think having 2 separate code paths can cause issues with normalization options (for example with `normalization = :pdf` weights should be `1` instead of `values[][1]`).

Comment (SimonDanisch) [reply src/stats/hist.jl]: Good idea, thanks :)

File: src/stats/hist.jl [hunk 2]
```diff
@@ -81,8 +85,8 @@ function Makie.plot!(plot::Hist)
             return bins
         end
     end
-
     points = lift(edges, plot.normalization, plot.scale_to) do edges, normalization, scale_to
+
```
Comment (piever) [inline src/stats/hist.jl:89]:
```suggestion

    points = lift(edges, plot.normalization, plot.scale_to) do edges, normalization, scale_to
```

Just to not add unrelated formatting changes to the PR

**Review by piever — APPROVED** [review]
